logstash-core/lib/logstash/util/cloud_setting_id.rb

```diff
@@ -0,0 +1,37 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "base64"
+
+module LogStash module Util class CloudSettingId
+  attr_reader :original, :decoded, :label, :elasticsearch_host, :kibana_host
+
+  def initialize(value)
+    return if value.nil?
+
+    unless value.is_a?(String)
+      raise ArgumentError.new("Cloud Id must be String. Received: #{value.class}")
+    end
+    @original = value
+    @label, sep, last = value.partition(":")
+    if last.empty?
+      @decoded = Base64.urlsafe_decode64(@label) rescue ""
+      @label = ""
+    else
+      @decoded = Base64.urlsafe_decode64(last) rescue ""
+    end
+    unless @decoded.count("$") == 2
+      raise ArgumentError.new("Cloud Id does not decode. Received: \"#{@original}\".")
+    end
+    cloud_host, es_server, kb_server = @decoded.split("$")
+    @elasticsearch_host = sprintf("%s.%s:443", es_server, cloud_host)
```

Comment:
https://github.com/elastic/beats/pull/4964/files#diff-1b409b9020aa8694f3beaeec8ee7ff72R113 uses :443, so I think this is ok.